### PR TITLE
Add docstrings to rest.py (#864)

### PR DESCRIPTION
**Pull request by ibmkendrick**

### CODE REVIEW

**Comment by ddebrunner** [comment]
Can one of the admins verify this patch?

File: com.ibm.streamsx.topology/opt/python/packages/streamsx/rest.py [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
-    """Creates a connection to a running Streams instance and exposes methods to retrieve the state of that instance.
+    """Creates a connection to a running distributed IBM Streams instance and exposes methods to retrieve the state of
+    that instance.
 
     Streams maintains information regarding the state of its resources. For example, these resources could include the
-    currently running Jobs, Views, PEs, Operators, and Domains. The StreamsConnection provides methods to retrieve that
-    information.
-
-    ``StreamsConnection`` connects to  distributed IBM Streams instance.
+    currently running Jobs, Views, PEs, Operators, and Domains. The :class:`StreamsConnection` provides methods to
```
Comment (ddebrunner) [inline com.ibm.streamsx.topology/opt/python/packages/streamsx/rest.py]: Curious, does `:class:` work? 

I thought it had to be `:py:class:` ?

Comment (ibmkendrick) [inline com.ibm.streamsx.topology/opt/python/packages/streamsx/rest.py]: Ya, it did work.  But I didn't read anything in Sphinx documentation saying that :py is the default domain, so maybe it's better to put it in?

**Comment by ibmkendrick** [comment]
Latest code addresses:
* change `:class` to `:py:class`
* document docstring conventions in DEVELOPMENT.md

**Comment by ddebrunner** [comment]
test now

**Comment by ibmkendrick** [comment]
Looks like I did manage to break the unit test with this change.
Since I changed the parameter name for StreamsConnection.get_instance() from `id` to `instance_id`, and I did not update tester(line 571) to do the same.

**Comment by ddebrunner** [comment]
Why was the `instance_id` changed to `id`?

**Pull request merged**